src/main/java/gregtech/api/nuclear/fission/FissionReactor.java

```diff
@@ -0,0 +1,724 @@
+            channel.computeWeightFromFuelRodMap();
+            if (channel.getWeight() > 0) {
+                effectiveCoolantChannels.add(channel);
+            }
+        }
+        CoolantChannel.normalizeWeights(effectiveCoolantChannels);
+    }
+
+    public boolean isDepleted() {
+        return fuelDepletion >= maxFuelDepletion || fuelDepletion < 0;
+    }
+
+    public void resetFuelDepletion() {
+        this.fuelDepletion = 0;
+    }
+
+    public void prepareInitialConditions() {
+        coolantBaseTemperature = 0;
+        coolantBoilingPointStandardPressure = 0;
+        coolantExitTemperature = 0;
+        coolantHeatOfVaporization = 0;
+        maxFuelDepletion = 0;
+        weightedGenerationTime = 0;
+
+        for (FuelRod rod : fuelRods) {
+            maxFuelDepletion += rod.getDuration();
+            weightedGenerationTime += rod.getNeutronGenerationTime();
+        }
+        if (fuelDepletion < 0) {
+            fuelDepletion = maxFuelDepletion;
+        }
+        weightedGenerationTime /= fuelRods.size();
+
+        for (CoolantChannel channel : coolantChannels) {
+            ICoolantStats prop = channel.getCoolant();
+            Fluid fluid = CoolantRegistry.originalFluid(prop);
+
+            if (fluid != null) {
+                coolantBaseTemperature += fluid.getTemperature();
+            }
+            coolantBoilingPointStandardPressure += prop.getBoilingPoint();
+            coolantExitTemperature += prop.getHotCoolant().getTemperature();
+            coolantHeatOfVaporization += prop.getHeatOfVaporization();
+        }
+
+        if (!coolantChannels.isEmpty()) {
+            coolantBaseTemperature /= coolantChannels.size();
+            coolantBoilingPointStandardPressure /= coolantChannels.size();
+            coolantExitTemperature /= coolantChannels.size();
+            coolantHeatOfVaporization /= coolantChannels.size();
+
+            if (coolantBaseTemperature == 0) {
+                coolantBaseTemperature = envTemperature;
+            }
+            if (coolantBoilingPointStandardPressure == 0) {
+                coolantBoilingPointStandardPressure = airBoilingPoint;
+            }
+        }
+        isOn = true;
+    }
+
+    /**
+     * Consumes the coolant. Calculates the heat removed by the coolant based on an amalgamation of different equations.
+     * It is not particularly realistic, but allows for some fine-tuning to happen. Heat removed is proportional to the
+     * surface area of the coolant channel (which is equivalent to the reactor's depth), as well as the flow rate of
+     * coolant and the difference in temperature between the reactor and the coolant
+     */
+    public double makeCoolantFlow(int flowRate) {
+        double heatRemoved = 0;
+        coolantMass = 0;
+        for (CoolantChannel channel : coolantChannels) {
+            FluidStack tryFluidDrain = channel.getInputHandler().getFluidTank().drain(flowRate, false);
```

Comment:
Could create variables here for the input and output tanks, so you don't have to constantly call `channel.getInputHandler.getFluidTank()` or `channel.getOutputHandler.getFluidTank()`